tests/websocket_test.py

```diff
@@ -1,25 +1,33 @@
-# -*- coding: utf-8 -*-
 """test websockets
 
 :copyright: Copyright (c) 2023 RadiaSoft LLC.  All Rights Reserved.
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
+
+import os
 import pytest
 
 
 def setup_module(module):
-    import os
-
     os.environ.update(
-        SIREPO_FEATURE_CONFIG_UI_WEBSOCKET="1",
         SIREPO_FEATURE_CONFIG_API_MODULES="srunit_api",
     )
 
 
+pytestmark = pytest.mark.skipif(
+    os.getenv("SIREPO_FEATURE_CONFIG_UI_WEBSOCKET", "1") != "1",
```

Comment:
The [default in dev is True](https://github.com/radiasoft/sirepo/blob/47b6e649200dfcd5fd44630a20e690802d101a53/sirepo/feature_config.py#L216) so this is a denormalization of that state.

Reply:
How do you decide whether to denormalize vs get the value from feature_config?

Reply:
In this particular case this is a "pytestmark" which is "pre test run", i.e. nothing can be imported at this point. We are denormalizing a few things:

- SIREPO_FEATURE_CONFIG_UI_WEBSOCKET is a denormalization instead of calling feature_cfg. This is an "acceptable" denormalization, but still a denormalization.
- The "1" is a denormalization, because it is the default in feature_cfg.

This is a bootstrap problem. We need to start sirepo, but we can't start it until we know if it will be configured with websockets iwc the test should be skipped.

We could maybe use this:

`pytest.skip("unsupported configuration")`

I am not sure how that would interact with pykern.pkcli.test. Something to try, but not right now.